Extended error handling with proper http responses

wklm opened the pull request:
hey Sergio :)

While making a request for instance as an unauthorized user exception is thrown already in the decorator, what makes it tricky to handle. A dirty workaround for that would be to create another decorator only for catching the JWTAuthorizationError and put it before the flask-jwtauthorization decorator. Instead of that I extended the library with appropriate http responses what makes it easier to use, as no additional actions are required from the library user.

Cheers,
Wojtek

wklm committed:
Extended error handling with proper http responses

sergiofcasado commented:
You are completely right. I had it in mind as a possible improvement, but never gave it a proper thought. 

I generally like your approach, but it raises some questions. What if we would like to return different error codes within the same block? Even more, what's the point now of having a exception which is raised and caught in the same code block? Maybe instead of raising the exception it is better to directly issue the abort instruction with appropriate error code and message.
This way we have fine grain control over returned codes and we clean up the code a bit. 

What do you think?

wklm commented:
It's rather just a matter of taste.

This:
```python 
    try:
        if parts[0].lower() != auth_bearer_prefix.lower():
            raise JWTAuthorizationError('Invalid JWT header', 'Unsupported authorization type')
        elif len(parts) == 1:
            raise JWTAuthorizationError('Invalid JWT header', 'Token missing')
        elif len(parts) > 2:
            raise JWTAuthorizationError('Invalid JWT header', 'Token contains spaces')
        if not parts[0]:
            raise JWTAuthorizationError('Authorization required', 'Request does not contain an access token')
        return parts[1]
    except JWTAuthorizationError as e:
        abort(400, e)
```

would be turned into this:

```python
    if parts[0].lower() != auth_bearer_prefix.lower():
        abort(400, ('Invalid JWT header', 'Unsupported authorization type'))
    elif len(parts) == 1:
        abort(400, ('Invalid JWT header', 'Token missing'))
    elif len(parts) > 2:
        abort(400, ('Invalid JWT header', 'Token contains spaces'))
    if not parts[0]:
        abort(400, ('Authorization required', 'Request does not contain an access token'))
    return parts[1]
```
Your approach is  better readable, but the current one provides more flexibility and will be eventually easier to extend in the future. Since we have a single potential error code for every function it makes sense for me, but if you opt for clarity over extensibility I can change it quickly.

wklm committed:
immediate abort instead of throwing exception

wklm committed:
remove spare blank lines

wklm committed:
remove JWTAuthorizationError exception

sergiofcasado committed:
Corrected conditionals for the pull request.

According to PEP8: https://www.python.org/dev/peps/pep-0008/#programming-recommendations
Second bullet point. I was doing it wrongly with != operator, but with your approach could have undesired behaviour.

sergiofcasado reviewed (approved):
Everything seems good to me.
